### ISSUE TITLE
Memory leak in FileUtils.loadBarEntriesFromAssets

### ISSUE THREAD

**Issue by farrukhnajmi** [PhilJay/MPAndroidChart#3989]
- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) file before posting this issue.

### The problem

The bug is a memory leak. At line 237 in https://github.com/PhilJay/MPAndroidChart/blob/master/MPChartLib/src/main/java/com/github/mikephil/charting/utils/FileUtils.java resource is allocated but not closed in case of Exception handler at line 250.

### What should happen

Code should use [try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) and make sure resource is closed in all cases.

**Comment by MGaetan89** [comment]
Unless I'm missing something, the resource is closed in the `finally` block, which is called even if an exception is thrown.
So there should not be a memory leak here, right?

**Comment by farrukhnajmi** [comment]
You are correct. I misread the code. The reason for filing this issue is that it was reported in a veracode static security scan of my app. 

Looking at current master version I do not see an obvious issue. I am using v3.0.2. I looked at history and the finally block was added in 5e2fb7bf9b007d1ad59e9fdbd79ceb0c42192082 as it is in latest version. I will find out from veracode why they thing there is an issue at line 237. Please leave issue open until I get an answer. Thanks.
